doc/CMakeLists.txt

```diff
@@ -2,50 +2,48 @@ file(GLOB json_files
     COMMAND jsonlint ${json_file})
 endforeach()
 
-set(BASELINE ${CMAKE_SOURCE_DIR}/doc/)
-set(COMMON_SCHEMA ${BASELINE}/common-schema.json)
-set(SR_COMMON_SCHEMA ${BASELINE}/sr-common-schema.json)
-
 set(AJV_OPTIONS --all-errors)
 
-get_filename_component(json_name common-schema.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv compile -s ${BASELINE}/${json_name}.json)
-
-get_filename_component(json_name sr-common-schema.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv compile -s ${BASELINE}/${json_name}.json
-  -r ${BASELINE}/common-schema.json)
-
-get_filename_component(json_name seg-schema.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv compile -s ${BASELINE}/${json_name}.json
-                                                   -r ${COMMON_SCHEMA})
-
-get_filename_component(json_name pm-schema.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv compile -s ${BASELINE}/${json_name}.json
-                                                   -r ${COMMON_SCHEMA})
-
-get_filename_component(json_name sr-tid1500-schema.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv compile -s ${BASELINE}/${json_name}.json
-                                                   -r ${COMMON_SCHEMA}
-                                                   -r ${SR_COMMON_SCHEMA})
-
-get_filename_component(json_name seg-example.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv -s ${BASELINE}/seg-schema.json
-  -r ${COMMON_SCHEMA}
-  -d ${BASELINE}/${json_name}.json)
-
-get_filename_component(json_name pm-example.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv -s ${BASELINE}/pm-schema.json
-  -r ${COMMON_SCHEMA}
-  -d ${BASELINE}/${json_name}.json)
-
-get_filename_component(json_name sr-tid1500-example.json NAME_WE)
-add_test(NAME ajv_${json_name} COMMAND ajv -s ${BASELINE}/sr-tid1500-schema.json
-  -r ${COMMON_SCHEMA}
-  -r ${SR_COMMON_SCHEMA}
-  -d ${BASELINE}/${json_name}.json ${AJV_OPTIONS})
+foreach(json_common_schema_file ${json_common_schema_files})
+  # -r needs to be used when specifying a schema that has references to external files
+  list(APPEND REF_SCHEMATA -r ${json_common_schema_file})
+endforeach()
+
+foreach(json_common_schema_file ${json_common_schema_files})
+  get_filename_component(json_name ${json_common_schema_file} NAME_WE)
+  add_test(NAME ajv_${json_name}
+    COMMAND ajv compile -s ${json_common_schema_file} ${REF_SCHEMATA} ${AJV_OPTIONS})
+endforeach()
+
+foreach(json_schema_file ${json_schema_files})
+  get_filename_component(json_name ${json_schema_file} NAME_WE)
+  add_test(NAME ajv_${json_name}
+           COMMAND ajv compile -s ${json_schema_file} ${REF_SCHEMATA} ${AJV_OPTIONS})
+endforeach()
+
+foreach(json_example_file ${json_example_files})
+  get_filename_component(json_name ${json_example_file} NAME_WE)
+  STRING(REGEX REPLACE -example.json -schema.json json_schema ${json_example_file})
+  add_test(NAME ajv_${json_name}
+    # -d stands for data that needs to be validated
```

Comment:
:) what I meant is to add a comment for each group of tests, explaining what is going on, not just a comment on one flag!

looks great otherwise!